src/Internal/Payment/Payment.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * Class Payment
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\Payment;
+
+use WC_Order;
+use WCPay\Internal\Payment\PaymentMethod\PaymentMethodInterface;
+
+/**
+ * Payment object.
+ *
+ * This is a data object, used as context and storage for payment states.
+ */
+class Payment {
```

Comment:
Yes, as far as states are concerned, the `Payment` object is their context. However, this doesn't mean that it's just a concept when looking at it globally.

Naming this `Payment` instead of `PaymentContext` is intentional. If we have a `PaymentState` and a `PaymentContext`, it sounds like there should be another class called `Payment` to link them, but there would be none.